commit 06b514847fffb79985e4bcde9f8dcc685c5d9ac5
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1328040359 -0600

    Fix konq filter in list view mode
    Select all now only selects shown (filtered) items

diff --git a/tdeui/klistview.cpp b/tdeui/klistview.cpp
index e1c5811..b52fe68 100644
--- a/kdeui/klistview.cpp
+++ b/kdeui/klistview.cpp
@@ -2356,6 +2356,52 @@ void KListViewItem::paintCell(TQPainter *p, const TQColorGroup &cg, int column,
   TQListViewItem::paintCell(p, _cg, column, width, tqalignment);
 }
 
+/*!
+    If \a select is TRUE, all the items get selected; otherwise all
+    the items get unselected. This only works in the selection modes \c
+    Multi and \c Extended. In \c Single and \c NoSelection mode the
+    selection of the current item is just set to \a select.
+*/
+
+void KListView::selectAll( bool select )
+{
+    if ( selectionMode() == Multi || selectionMode() == Extended ) {
+	bool b = signalsBlocked();
+	blockSignals( TRUE );
+	bool anything = FALSE;
+	TQListViewItemIterator it( this );
+	while ( it.current() ) {
+	    TQListViewItem *i = it.current();
+	    if ( select == TRUE ) {
+		if ( (bool)i->isVisible() == TRUE ) {
+		    i->setSelected( TRUE );
+		    anything = TRUE;
+		}
+		if ( (bool)i->isVisible() == FALSE ) {
+		    i->setSelected( FALSE );
+		    anything = TRUE;
+		}
+	    }
+	    else {
+		if ( (bool)i->isSelected() != select ) {
+		    i->setSelected( select );
+		    anything = TRUE;
+		}
+	    }
+	    ++it;
+	}
+	blockSignals( b );
+	if ( anything ) {
+	    emit selectionChanged();
+// 	    d->useDoubleBuffer = TRUE;
+	    triggerUpdate();
+	}
+    } else if ( currentItem() ) {
+	TQListViewItem * i = currentItem();
+	setSelected( i, select );
+    }
+}
+
 void KListView::virtual_hook( int, void* )
 { /*BASE::virtual_hook( id, data );*/ }
 
diff --git a/tdeui/klistview.h b/tdeui/klistview.h
index ed10192..c8bb837 100644
--- a/kdeui/klistview.h
+++ b/kdeui/klistview.h
@@ -660,6 +660,14 @@ public slots:
    */
   bool tabOrderedRenaming() const;
 
+  /**
+   * Override TQListView selectAll() so that filtered
+   * items are not selected
+   *
+   * @since 14.0
+   */
+  virtual void selectAll( bool select );
+
 protected:
   /**
    * Determine whether a drop on position @p p would count as
